runtime-modules/content/src/lib.rs

```diff
@@ -1417,21 +1460,51 @@ decl_module! {
             // Authorize nft owner
             ensure_actor_authorized_to_manage_nft::<T>(origin, &owner_id, &nft.owner, video.in_channel)?;
 
-            // Ensure auction for given video id exists
-            let auction = nft.ensure_auction_state::<T>()?;
+            // Ensure auction state that can be canceled
+            Self::ensure_english_auction_state(&nft)
+                .and_then(|eng| eng.ensure_auction_can_be_canceled::<T>())?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Cancel auction
+            let nft = Self::cancel_transaction(&nft, None);
+
+            VideoById::<T>::mutate(video_id, |v| v.set_nft_status(nft));
 
-            // Ensure given auction can be canceled
-            auction.ensure_auction_can_be_canceled::<T>()?;
+            // Trigger event
+            Self::deposit_event(RawEvent::AuctionCanceled(owner_id, video_id));
+        }
+
+        // Cancel video nft english auction
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn cancel_open_auction(
```

Comment:
This is incorrectly blocked by not having bids... can be done at any time.

Reply:
Removed here e256eba1394bd5ce6fe5bde99645c90cde5b36e8